src/harness/fourslash.ts

```diff
@@ -178,6 +179,20 @@ module FourSlash {
                     case metadataOptionNames.sourceRoot:
                         settings.sourceRoot = globalOptions[prop];
                         break;
+                    case metadataOptionNames.jsx:
+                        switch (globalOptions[prop].toLowerCase()) {
+                            case "react":
+                                settings.jsx = ts.JsxEmit.React;
+                                break;
+                            case "preserve":
+                                settings.jsx = ts.JsxEmit.Preserve;
+                                break;
+                            default:
+                                ts.Debug.assert(globalOptions[prop] === undefined || globalOptions[prop] === "None");

```

Comment:
If `globalOptions[prop]` is `undefined`, then `globalOptions[prop].toLowerCase()` above will crash.